app/models/piece.rb

```diff
@@ -6,10 +6,23 @@ def self.types
     %w[King Queen Rook Bishop Knight Pawn]
   end
 
-  def self.colors
+  def self.color
     %w[White Black]
   end
 
+  def move_to!(piece, destination_x, destination_y)
+    destination_piece_check = Piece.find_by(column_coordinate: destination_x, row_coordinate: destination_y).present?
```

Comment:
Also.

Performance tip: you can use `exists?`

```
.find_by(column_coordinate: destination_x, row_coordinate: destination_y).present?

# change to.

.find_by(column_coordinate: destination_x, row_coordinate: destination_y).exists?
```

The `exists?` version executes a faster query.